munit/shared/src/test/scala/hedgehog/munit/HedgehogAssertionsSuite.scala

```diff
@@ -0,0 +1,790 @@
+
+  val instanceLogNameExamplesAndComparisonFunction =
+    FunFixture.map2(instanceExamplesAndComparisonFunction, logName)
+
+  val cond = FunFixture[Boolean](setup = test => true, teardown = suite => ())
+
+  val failCond =
+    FunFixture[Boolean](setup = test => false, teardown = suite => ())
+
+  val clue = FunFixture[String](setup = test => "clue", teardown = suite => ())
+
+  val instanceConditionAndClue = FunFixture.map3(instance, cond, clue)
+
+  val instanceConditionClueAndSuccess =
+    FunFixture.map2(instanceConditionAndClue, resultSuccess)
+
+  val failException = FunFixture[Exception](
+    setup = test => new Exception("fail!"),
+    teardown = suite => ()
+  )
+
+  val instanceAndSuccess = FunFixture.map2(instance, resultSuccess)
+
+  val instanceAndFailException = FunFixture.map2(instance, failException)
+
+  val log = FunFixture[Log](
+    setup = test => Log.String2Log("woot"),
+    teardown = suite => ()
+  )
+
+  val instanceAndLog = FunFixture.map2(instance, log)
+
+  val instanceConditionAndSuccess =
+    FunFixture.map3(instance, cond, resultSuccess)
+
+  val instanceConditionAndFailure =
+    FunFixture.map3(instance, failCond, resultFailure)
+
+  val instanceAndExample1AndClueAndSuccess =
+    FunFixture.map2(FunFixture.map3(instance, exampleOne, clue), resultSuccess)
+
+  val failResult = FunFixture[Result](
+    setup = test =>
+      Result.failure
+        .log("=== Failed ===")
+        .log("--- lhs ---")
+        .log("1")
+        .log("--- rhs ---")
+        .log("2"),
+    teardown = suite => ()
+  )
+
+  val instanceExamplesAndClueAndFailureExamples =
+    FunFixture.map3(
+      FunFixture.map3(instance, exampleOne, exampleTwo),
+      clue,
+      failResult
+    )
+
+  val instanceExamplesAndSuccess =
+    FunFixture.map3(instance, exampleOne, resultSuccess)
+
+  val instanceExamplesAndFailure = FunFixture.map2(
+    FunFixture.map3(instance, exampleOne, exampleTwo),
+    failResult
+  )
+
+  val doubleExample =
+    FunFixture[Double](setup = testOptions => 1.00, teardown = _ => ())
+
+  val doubleExample2 =
+    FunFixture[Double](setup = testOptions => 2.00, teardown = _ => ())
+
+  val doubleExample3 =
+    FunFixture[Double](setup = testOptions => 3.00, teardown = _ => ())
+
+  val instanceDoubleExampleClueAndSuccess = FunFixture.map2(
+    FunFixture.map3(instance, doubleExample, clue),
+    resultSuccess
+  )
+
+  val deltaDouble =
+    FunFixture[Double](setup = testOptions => 1.00, teardown = _ => ())
+
+  val instanceDoubleExamplesDeltaClueAndSuccess = FunFixture.map2(
+    FunFixture.map3(
+      FunFixture.map3(instance, doubleExample, doubleExample2),
+      deltaDouble,
+      clue
+    ),
+    resultSuccess
+  )
+
+  val resultDeltaFailure = FunFixture[Result](
+    setup = testOptions => {
+      Result.failure
+        .log("=== Failed ===")
+        .log("--- lhs ---")
+        .log("1.0")
+        .log("--- rhs ---")
+        .log("3.0")
```

Comment:
I think this one cause [the Scala.js build failure](https://github.com/hedgehogqa/scala-hedgehog/runs/6440589170?check_suite_focus=true#step:6:272).
In Scala.js build, it looks like the actual values are `1` and `3` instead of `1.0` and `3.0`.

You may have a solution for this, but one solution I can think is
1. Remove `resultDeltaFailure` from `HedgehogAssertionsSuite`.
1. Add `src/test/scala` to both `js` and `jvm` projects.
1. Add the `hedgehog.munit` package to `src/test/scala`
1. Create a `trait` containing `resultDeltaFailure`.
   e.g.)
   JVM version:
   ```scala
   package hedgehog.munit
   
   import _root_.munit.FunSuite
   import hedgehog.Result
   
   trait FailureFixtures {
     self: FunSuite =>
   
     val resultDeltaFailure = FunFixture[Result](
       setup = testOptions => {
         Result.failure
           .log("=== Failed ===")
           .log("--- lhs ---")
           .log("1.0")
           .log("--- rhs ---")
           .log("3.0")
       },
       teardown = _ => ()
     )
   
   }
   ```
   JS version:
   ```scala
   package hedgehog.munit
   
   import _root_.munit.FunSuite
   import hedgehog.Result
   
   trait FailureFixtures {
     self: FunSuite =>
   
     val resultDeltaFailure = FunFixture[Result](
       setup = testOptions => {
         Result.failure
           .log("=== Failed ===")
           .log("--- lhs ---")
           .log("1")
           .log("--- rhs ---")
           .log("3")
       },
       teardown = _ => ()
     )
   
   }
   ```
1. Use it with `HedgehogAssertionsSuite`
   e.g.)
   ```scala
   class HedgehogAssertionsSuite extends FunSuite with FailureFixtures {
   ```
   (If you use IntelliJ IDEA, it may not know about `FailureFixtures` but it compiles and has no issue.

Reply:
Once it's done, you sill have unused param errors in Scala 2.12.
Let me think about how to fix it. There is another error from compiler flags in Scala 3 as well.

Reply:
@Kevin-Lee -- All of the build failures are addressed in https://github.com/hedgehogqa/scala-hedgehog/pull/229/commits/77778f264a524787f9ff6a71f08db368e2f1acf3

Not seeing any other compiler flag errors.

Reply:
scala-js tests re-added with https://github.com/hedgehogqa/scala-hedgehog/pull/229/commits/4dbc441e2ae3f93e498f1d04949b70c1d3990db6